### PR TITLE
Merge rogii-v3.4.0-dev

### PR DESCRIPTION
**Pull request by Shkarlatov**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -51,17 +52,78 @@ endif()
...
+
+include(${ROGII_FOLDER_PATH}/Version.cmake)
+include(${ROGII_FOLDER_PATH}/npm/Include.cmake)
+include(${ROGII_FOLDER_PATH}/environment.cmake)
+
+include(${ROGII_FOLDER_PATH}/utils/SetupRpathLink.cmake)
+include(${ROGII_FOLDER_PATH}/utils/Rpath.cmake)
+
+if(OSG_QT_ANGLE AND MSVC)
+	CNPM_ADD_PACKAGE(
```
Comment (chak3r) [inline CMakeLists.txt]: If I remember correctly, we only need Qt if OSG_QT_ANGLE is set and only for Windows. But we need msvs and WindowsSDK for Windows and gxx_runtime for linux anyway. Also you have these definitions in environment_linux.cmake and environment_windows.cmake. Please put things in order.

File: rogii/build_common.cmake [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+execute_process(
+    COMMAND
+        "${CMAKE_COMMAND}" --build . -j 5
+    WORKING_DIRECTORY
+        "${RELEASE_PATH}"
+)
+
+execute_process(
+    COMMAND
+        "${CMAKE_COMMAND}" --build . -j 5 --target install
```
Comment (chak3r) [inline rogii/build_common.cmake]: Please remove -j 5. It was my mistake.

File: rogii/build_common.cmake [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+endif()
+
+execute_process(
+    COMMAND
+        "${CMAKE_COMMAND}" -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX} -DOSG_MSVC_VERSIONED_DLL=OFF -DBUILD_OSG_APPLICATIONS=OFF ${ANGLE} ${GENERATOR} ../..
+    WORKING_DIRECTORY
+        "${DEBUG_PATH}"
+)
+
+execute_process(
```
Comment (chak3r) [inline rogii/build_common.cmake]: `"${CMAKE_COMMAND}" --build . --target install` is enough. I think we can remove this execute_process.

File: rogii/Version.cmake [hunk 4]
```diff
@@ -0,0 +1,6 @@
+SET(OPENSCENEGRAPH_MAJOR_VERSION 3)
```
Comment (chak3r) [inline rogii/Version.cmake]: Please rename file to lowercase.

**Review by chak3r — COMMENTED** [review]
In general, I'm not ready to approve this request yet. Many strange changes in osg cmakelists files. I will try to see who made it and figure out why they are needed.

File: rogii/environment_linux.cmake [hunk 5]
```diff
@@ -0,0 +1,4 @@
+set(
+    REPOSITORY_OS
+    "linux"
+)
```
Comment (chak3r) [inline rogii/environment_linux.cmake]: We need gxx_runtime package to checking compiler under linux.
```
CNPM_ADD_PACKAGE(
    NAME
        gxx_runtime
    VERSION
        9.2.1
    BUILD_NUMBER
        0
)
```

Comment (Shkarlatov) [reply rogii/environment_linux.cmake]: Added

File: CMakeLists.txt [hunk 6]
```diff
@@ -51,17 +52,33 @@ endif()
...
-SET(OPENSCENEGRAPH_MINOR_VERSION 4)
-SET(OPENSCENEGRAPH_PATCH_VERSION 0)
-SET(OPENSCENEGRAPH_SOVERSION 130)
+
+set(
+    ROGII_FOLDER_PATH
+    "${CMAKE_CURRENT_LIST_DIR}/rogii"
+)
+
+include(${ROGII_FOLDER_PATH}/Version.cmake)
```
Comment (mr-bit) [inline CMakeLists.txt]: Rename file Version.cmake to version.cmake and use this lowercased name everywhere

File: rogii/build_common.cmake [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+
+set(
+    CMAKE_INSTALL_PREFIX
+    ${ROOT}/${PACKAGE_NAME}
+)
+
+if(WIN32)
+    set(
+        GENERATOR
+        -G Ninja
```
Comment (mr-bit) [inline rogii/build_common.cmake]: Use Ninja generator for all platforms, this condition is old and not actual

File: rogii/environment_linux.cmake [hunk 8]
```diff
@@ -0,0 +1,13 @@
+set(
+    REPOSITORY_OS
```
Comment (mr-bit) [inline rogii/environment_linux.cmake]: Unused and not actual, remove this variable

File: rogii/environment_windows.cmake [hunk 9]
```diff
@@ -0,0 +1,20 @@
+include(${CMAKE_CURRENT_LIST_DIR}/msvs_package.cmake)
+include(${CMAKE_CURRENT_LIST_DIR}/windowssdk_package.cmake)
+
+set(
+    REPOSITORY_OS
```
Comment (mr-bit) [inline rogii/environment_windows.cmake]: Unused and not actual, remove this variable

File: rogii/environment_windows.cmake [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+)
+
+if(OSG_QT_ANGLE)
+    CNPM_ADD_PACKAGE(
+        NAME
+            Qt
+        VERSION
+            5.15.1
+        BUILD_NUMBER
+            0
```
Comment (mr-bit) [inline rogii/environment_windows.cmake]: Change to actual BUILD_NUMBER 1

**Review by mr-bit — CHANGES_REQUESTED** [review]
Good but small changes are required

File: rogii/build_common.cmake [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+if(WIN32)
+    set(
+        ANGLE
+        -DOSG_QT_ANGLE=ON
+    )
+endif()
+
+execute_process(
+    COMMAND
+        "${CMAKE_COMMAND}" -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX} -DOSG_MSVC_VERSIONED_DLL=OFF -DBUILD_OSG_APPLICATIONS=OFF ${ANGLE} ${GENERATOR} ../..
```
Comment (mr-bit) [inline rogii/build_common.cmake]: Remove unused ${GENERATOR}

File: rogii/build_common.cmake [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+)
+
+file(
+    MAKE_DIRECTORY
+    "${RELEASE_PATH}"
+)
+
+execute_process(
+    COMMAND
+        "${CMAKE_COMMAND}" -G Ninja -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX} -DOSG_MSVC_VERSIONED_DLL=OFF -DBUILD_OSG_APPLICATIONS=OFF -DCMAKE_RELWITHDEBINFO_POSTFIX= ${ANGLE} ${GENERATOR} ../..
```
Comment (mr-bit) [inline rogii/build_common.cmake]: Remove unused ${GENERATOR}

**Review by mr-bit — CHANGES_REQUESTED** [review]
Last changes

**Review by chak3r — APPROVED** [review]

**Review by mr-bit — APPROVED** [review]

**Pull request merged**